exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -38,8 +44,12 @@
 // methods for compatibility.)
 class DrillStatementImpl extends AvaticaStatement implements DrillStatement,
                                                              DrillRemoteStatement {
+  //Not using the DrillbitContext's ExecutorService as this is threadPool is light-weight (threads wake up to cancel tasks) but needs a low response time
+  private static ExecutorService queryTimeoutTaskPool = Executors.newCachedThreadPool(new NamedThreadFactory("q-timeout-"));
```

Comment:
Using a threadPool model is actually making the solution simpler, since I'm triggering the query cancellation. The TimeoutTrigger class uses an executorService for this cleanly, without exposing anything else. Checking constantly for the remaining time seemed like an overhead that would be cumulatively add to the runtime. 
The one benefit of doing this in the Cursor would be that I could, potentially, have a new QueryState defined as TIMED_OUT, but I'm reluctant to modify the Protobuf as of now.